app/src/main/java/com/trevjonez/daggertest/main_activity/MainActivity.java

```diff
@@ -16,22 +16,57 @@
 
 package com.trevjonez.daggertest.main_activity;
 
+import android.app.Fragment;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
+import android.widget.TextView;
 
 import com.trevjonez.daggertest.R;
+import com.trevjonez.daggertest.fragment.fragment_one.FragmentOne;
+import com.trevjonez.daggertest.fragment.fragment_two.FragmentTwo;
+import com.trevjonez.inject.PlainComponent;
 import com.trevjonez.inject.activity.ActivityComponentBuilderHost;
+import com.trevjonez.inject.fragment.FragmentComponentBuilder;
+import com.trevjonez.inject.fragment.FragmentComponentBuilderHost;
 
-public class MainActivity extends AppCompatActivity {
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+
+public class MainActivity extends AppCompatActivity implements FragmentComponentBuilderHost{
+    TextView textView;
 
     MainActivityComponent component;
+    @Inject
+    Map<Class<? extends Fragment>, Provider<FragmentComponentBuilder>> fragmentComponentBuilderMap;
+    @Inject @Named("TextMainActivity") String textMainActivity;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        textView= (TextView) findViewById(R.id.textView_mainActivity);
 
-        component = ((ActivityComponentBuilderHost) getApplication()).getActivityComponentBuilder(MainActivity.class, MainActivityComponent.Builder.class).build();
+        component = ((ActivityComponentBuilderHost) getApplication())
```

Comment:
This is my bad since It was wrong before but will you please move the component creation and injection to happen before calling `setContentView(int)`?

This is because fragments can be included via xml and not always via in code fragment transactions. If we set content view first than any fragment included in the xml will cause a NPE when calling  `getFragmentComponentBuilder`